### ISSUE TITLE
MIDI export exports everything as piano, making things out of tune

### ISSUE THREAD

**Issue by musikBear** [LMMS/lmms#5287]
The attached project was given to me, because it could not be exported.
I have tried to export with 1.2.1 
It exports 51kb, and cuts off almost exactly one mins.
I have shaved it down to minimum, and added a control beeper at project end.
I also had to add an envelope to watsyn, due to clicks and pops
[crash.zip](https://github.com/LMMS/lmms/files/3784388/crash.zip)

( I have NO idea what this music is called...

I then took the demo 'Kaakkuja' and midi-exported that
[Greippi - Krem Kaakkuja (Second Flight Remix).zip](https://github.com/LMMS/lmms/files/3784622/Greippi.-.Krem.Kaakkuja.Second.Flight.Remix.zip)
Its is fully exported, but completely broken

**Comment by PhysSong** [comment]
I can't see any issues with the MIDI export. @musikBear Could you provide more details?

**Comment by musikBear** [comment]
> I can't see any issues with the MIDI export.

Kaakkuja?!
Cant you hear that it is complete out of synch and tuning? What midi-player do you have? Mine is Anvil on Windows64.
If that project sound fine on your midi.player it would be great, because then Anvil have the problem, not LMMS!
This is what i can hear
[Kaakkuja.zip](https://github.com/LMMS/lmms/files/3799019/Kaakkuja.zip)
Pretty much unbearable :[ @PhysSong 
**Also** -I cant **reimport** the MIDI-file *into* LMMS at all! Try that too!

How much of the project crash.zip could you export?

**Comment by SecondFlight** [comment]
Hi, I'm the author of the second song, and I can confirm everything there is working as expected.

The issue is that everything is exporting as piano, which includes drums. I'm using AFP for most of the drums in that song, and converting the drum notes directly to piano creates a bunch of awful off-tune stuff. It sounds to me like the non-drum notes are coming out correctly.

TL;DR: if the job of the MIDI export is to convert all piano roll stuff to actual piano, then it's working just fine.

Here's the song for reference:
https://soundcloud.com/second_flight/krem-kaakkuja-remix

**Comment by coderman64** [comment]
Hey folks! AFAIK, the MIDI support in LMMS is incomplete and is only meant to transfer tracks between DAWs. If you try to play any MIDI files exported in this fashion in some sort of media player, every instrument will be converted to the piano. Since MIDI files only support sound fonts, maybe it would be possible to support exporting MIDI files with correct instruments for all sf2 tracks, but I don't think this is currently supported. As far as the given file goes, I was able to export it at full length (though the instruments were still broken), and then reimport it into a single LMMS track without much issue. (This was using 1.2.1 on Windows 10).

**Comment by IkibiSction** [comment]
> Hey folks! AFAIK, the MIDI support in LMMS is incomplete and is only meant to transfer tracks between DAWs. If you try to play any MIDI files exported in this fashion in some sort of media player, every instrument will be converted to the piano. Since MIDI files only support sound fonts, maybe it would be possible to support exporting MIDI files with correct instruments for all sf2 tracks, but I don't think this is currently supported. As far as the given file goes, I was able to export it at full length (though the instruments were still broken), and then reimport it into a single LMMS track without much issue. (This was using 1.2.1 on Windows 10).

It's been years

**Comment by musikBear** [comment]
> It's been years

Aaand is one of the features that is considered in next release 
https://github.com/LMMS/lmms/pull/8048
You can even **test** it yourself
